syntax = "proto3";

service Booking {
  rpc GetListBookings (EmptyMessage) returns (stream Book);
  rpc GetListBookingsFromUser (UserId) returns (stream Book);
  rpc AddBooking (Reservation) returns (AddBookingReturnMessage);
}

message EmptyMessage{}

message Book {
  message Dates{
    string date = 1;
    repeated string movies = 2;
  }

  string userid = 1;
  repeated Dates dates = 2;
}

message UserId{
  string userid = 1;
}

message Reservation{
  string userid = 1;
  string movie = 2;
  string date = 3;
}

message AddBookingReturnMessage{
  string message = 1;
}